packages/tao/src/shared/tree.ts

```diff
@@ -255,6 +255,13 @@ export class FsTree implements Tree {
 
   private directChildrenOfDir(path: string): string[] {
     const res = {};
+    if (path === '') {
```

Comment:
I added tests for `children` after different actions to the tree as well as specific tests for the root of the tree. Apparently, directories were also not being cleaned up when all files were removed. 

`tree.children` wasn't working properly when the root was passed.

e.g.

```ts
tree.children(tree.root);
tree.children('');
```

The original logic checks if the changes start with `path + '/'` which when the root is passed in is just `startsWith('/')` which no change will start with since they are all normalized.